### PR TITLE
Use range based for-loop where applicable

### PR DESCRIPTION
**Pull request by feliwir**
- [x] add `.clang-tidy` file
- [x] Use range-based loops when we can:
  - Remove a magic number
  - Improve readability

### CODE REVIEW

**Comment by feliwir** [comment]
Can't do this with STLPort - wait for standalone

**Comment by duncanspumpkin** [comment]
Should work after a rebase now.

File: src/game/common/modules/modulefactory.cpp [hunk 1]
```diff
@@ -48,10 +48,10 @@ void ModuleFactory::Init()
  */
 void ModuleFactory::CRC_Snapshot(Xfer *xfer)
 {
-    for (auto it = m_moduleDataList.begin(); it != m_moduleDataList.end(); ++it) {
+    for (auto &it : m_moduleDataList) {
```
Comment (xezon) [inline src/game/common/modules/modulefactory.cpp:51]: Should this maybe have a better name than `it` ? For example `moduleData` ?

Comment (xezon) [inline src/game/common/modules/modulefactory.cpp:51]: Should this maybe have a better name than `it` ? For example `moduleData` ?

File: src/game/common/modules/modulefactory.cpp [hunk 2]
```diff
@@ -62,9 +62,9 @@ void ModuleFactory::CRC_Snapshot(Xfer *xfer)
  */
 void ModuleFactory::Xfer_Snapshot(Xfer *xfer)
 {
-    for (auto it = m_moduleDataList.begin(); it != m_moduleDataList.end(); ++it) {
+    for (auto &it : m_moduleDataList) {
```
Comment (xezon) [inline src/game/common/modules/modulefactory.cpp:65]: Should this maybe have a better name than `it` ? For example `moduleData` ?

Comment (xezon) [inline src/game/common/modules/modulefactory.cpp:65]: Should this maybe have a better name than `it` ? For example `moduleData` ?

File: src/game/common/rts/productionprerequisite.cpp [hunk 3]
```diff
@@ -149,8 +149,8 @@ Utf16String ProductionPrerequisite::Get_Requires_List(const Player *player) cons
 
         bool has_sciences = true;
 
-        for (size_t i = 0; i < m_prereqScience.size(); i++) {
-            if (!player->Has_Science(m_prereqScience[i])) {
+        for (auto reqScience : m_prereqScience) {
```
Comment (xezon) [inline src/game/common/rts/productionprerequisite.cpp:152]: Maybe just call it `science`

Comment (xezon) [inline src/game/common/rts/productionprerequisite.cpp:152]: Maybe just call it `science`

File: src/game/common/userpreferences.cpp [hunk 4]
```diff
@@ -61,8 +61,8 @@ bool UserPreferences::Write()
     FILE *fp = fopen(m_filename.Str(), "w");
 
     if (fp != nullptr) {
-        for (auto it = begin(); it != end(); ++it) {
-            fprintf(fp, "%s = %s\n", it->first.Str(), it->second.Str());
+        for (auto &it : *this) {
```
Comment (xezon) [inline src/game/common/userpreferences.cpp:64]: What is `it` ? Can this have a better name? If not please don't use `auto` type. It is very difficult to read from review what is going on here.

Comment (xezon) [inline src/game/common/userpreferences.cpp:64]: What is `it` ? Can this have a better name? If not please don't use `auto` type. It is very difficult to read from review what is going on here.

**Comment by feliwir** [comment]
I'll revisit this at a later stage

**Pull request closed without merging**